PR: https://github.com/odoo/odoo/pull/233157

From: 72db9ed7ef1540defafb5531eea260366120d160
From: tsm-odoo
Date: 2025-10-28 11:52:56

Breaking data model changes scores: del:8 + add:0, change matches:
-    livechat_conversation_tag_ids = fields.Many2many(
-    conversation_tag_ids = fields.Many2many(
-    name = fields.Char("Name", required=True)
-    color = fields.Integer("Color", default=_get_default_color)
-    conversation_ids = fields.Many2many(

Total Changes: 62

[REM] im_livechat: remove conversation tags

task-5190357

closes odoo/odoo#233157

Related: odoo/upgrade#8699
Signed-off-by: Matthieu Stockbauer (tsm) <tsm@odoo.com>

================================= pseudo patch: =================================

--- a/addons/im_livechat/models/__init__.py
+++ b/addons/im_livechat/models/__init__.py
@@ -15,7 +15,6 @@ from . import discuss_channel
 from . import discuss_channel_member
 from . import discuss_channel_rtc_session
 from . import im_livechat_channel_member_history
-from . import im_livechat_conversation_tag
 from . import mail_message
 from . import res_users_settings
 from . import rating_rating

--- a/addons/im_livechat/models/discuss_channel.py
+++ b/addons/im_livechat/models/discuss_channel.py
@@ -131,13 +131,6 @@ class DiscussChannel(models.Model):
         compute="_compute_livechat_outcome",
         store=True,
     )
-    livechat_conversation_tag_ids = fields.Many2many(
-        "im_livechat.conversation.tag",
-        "livechat_conversation_tag_rel",
-        groups="im_livechat.im_livechat_group_user",
-        string="Live Chat Conversation Tags",
-        help="Tags to qualify the conversation.",
-    )
     livechat_start_hour = fields.Float(
         "Session Start Hour", compute="_compute_livechat_start_hour", store=True
     )
@@ -412,13 +405,6 @@ class DiscussChannel(models.Model):
                 Store.Attr("livechat_note", predicate=is_livechat_channel),
                 Store.Attr("livechat_status", predicate=is_livechat_channel),
                 Store.Many("livechat_expertise_ids", ["name"], predicate=is_livechat_channel),
-                # sudo: internal users having access to the channel can read its tags
-                Store.Many(
-                    "livechat_conversation_tag_ids",
-                    ["name", "color"],
-                    predicate=is_livechat_channel,
-                    sudo=True,
-                ),
             ],
         )
         return field_names
@@ -456,13 +442,6 @@ class DiscussChannel(models.Model):
                     Store.Attr("livechat_outcome", predicate=is_livechat_channel),
                     Store.Attr("livechat_status", predicate=is_livechat_channel),
                     Store.Many("livechat_expertise_ids", ["name"], predicate=is_livechat_channel),
-                    # sudo: internal users having access to the channel can read its tags
-                    Store.Many(
-                        "livechat_conversation_tag_ids",
-                        ["name", "color"],
-                        predicate=is_livechat_channel,
-                        sudo=True,
-                    ),
                 ],
             )
         return super()._to_store_defaults(target) + fields

--- a/addons/im_livechat/models/im_livechat_channel_member_history.py
+++ b/addons/im_livechat/models/im_livechat_channel_member_history.py
@@ -32,11 +32,6 @@ class ImLivechatChannelMemberHistory(models.Model):
     agent_expertise_ids = fields.Many2many(
         "im_livechat.expertise", compute="_compute_member_fields", store=True
     )
-    conversation_tag_ids = fields.Many2many(
-        "im_livechat.conversation.tag",
-        "im_livechat_channel_member_history_conversation_tag_rel",
-        related="channel_id.livechat_conversation_tag_ids",
-    )
     avatar_128 = fields.Binary(compute="_compute_avatar_128")
 
     # REPORTING FIELDS

--- a/addons/im_livechat/models/im_livechat_conversation_tag.py
+++ b/None
@@ -1,35 +0,0 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from random import randint
-
-from odoo import fields, models, api, Command
-
-
-class Im_LivechatConversationTag(models.Model):
-    """Tags for Live Chat conversations."""
-
-    _name = "im_livechat.conversation.tag"
-    _description = "Live Chat Conversation Tag"
-    _order = "name"
-
-    @api.model
-    def _get_default_color(self):
-        return randint(1, 11)
-
-    name = fields.Char("Name", required=True)
-    color = fields.Integer("Color", default=_get_default_color)
-    conversation_ids = fields.Many2many(
-        "discuss.channel",
-        "livechat_conversation_tag_rel",
-        string="Discuss Channels",
-    )
-
-    _name_unique = models.UniqueIndex("(name)")
-
-    @api.ondelete(at_uninstall=False)
-    def _unlink_sync_conversation(self):
-        # For triggering the _sync_field_names before being unlinked
-        # sudo: users who can delete tags can remove them from conversations in cascade
-        self.sudo().conversation_ids.livechat_conversation_tag_ids = [
-            Command.unlink(tag.id) for tag in self
-        ]
